util/topsql/tracecpu/profile.go

```diff
@@ -62,123 +51,90 @@ type SQLCPUTimeRecord struct {
 	CPUTimeMs  uint32
 }
 
-type sqlCPUProfiler struct {
-	taskCh chan *profileData
+// SQLCPUCollector uses to consume cpu profile from globalCPUProfiler, then parse the SQL CPU usage from the cpu profile data.
+type SQLCPUCollector struct {
+	ctx    context.Context
+	cancel context.CancelFunc
+	wg     sync.WaitGroup
 
-	mu struct {
-		sync.Mutex
-		ept *exportProfileTask
-	}
-	collector atomic.Value
+	profileConsumer cpuprofile.ProfileConsumer
+	collector       Collector
 }
 
-var (
-	defaultProfileBufSize = 100 * 1024
-	profileBufPool        = sync.Pool{
-		New: func() interface{} {
-			return bytes.NewBuffer(make([]byte, 0, defaultProfileBufSize))
-		},
-	}
-)
-
-// newSQLCPUProfiler create a sqlCPUProfiler.
-func newSQLCPUProfiler() *sqlCPUProfiler {
-	return &sqlCPUProfiler{
-		taskCh: make(chan *profileData, 128),
+// NewSQLCPUCollector create a SQLCPUCollector.
+func NewSQLCPUCollector(c Collector) *SQLCPUCollector {
+	ctx, cancel := context.WithCancel(context.Background())
+	return &SQLCPUCollector{
+		ctx:             ctx,
+		cancel:          cancel,
+		profileConsumer: make(cpuprofile.ProfileConsumer, 1),
+		collector:       c,
 	}
 }
 
-func (sp *sqlCPUProfiler) Run() {
-	logutil.BgLogger().Info("cpu profiler started")
-	go sp.startCPUProfileWorker()
+// Start uses to start to run SQLCPUCollector.
+func (sp *SQLCPUCollector) Start() {
```

Comment:
How about reducing the API surface to only `Start` and `Stop`? The current API seems to be unsound - `disable` does not disable all data producing from `SQLCPUCollector`, and it only disables the CPU profiler. It is usually very rare that a module has `Start`, `Stop`, `Enable`, `Disable`.